### PR TITLE
#4255 - fixed + throwing an error when sent via graphql

### PR DESCRIPTION
**Pull request by florianf22**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/4255

**Problem:**
* When inputting the symbol '+' in the search, it would throw an error, a graphql one.

**In this PR:**
* Graphql treats the symbol '+' differently, most likely. To get a very similar results, modified a search term in the query class. Now, i'll replace all the '+' with '-' that graphql can handle (it'll have a very similar results)

### CODE REVIEW

**Comment by Elena-Evsa** [comment]
Still can see the errors if user enter + in search query
image

**Comment by florianf22** [comment]
Hi, looks like not all the files had been pushed, check now, please

**Comment by Elena-Evsa** [comment]
1. Sign + absent in header on mobile:
image

**Comment by AleksandrsKondratjevs** [comment]
`Now, i'll replace all the '+' with '-' that graphql can handle (it'll have a very similar results)`
What means very similar results? We need same result, not similar. Won't elastic in this case search for `short-jeans-for-man` instead of `short jeans for man` ?

**Comment by florianf22** [comment]
Before I made any change, '+' was parsed anyway. Graphql doesn't want to get '+'. If you, for example search for 'short+jeans', it returns 107 results. if you do 'short jeans', again 107 results. 'short-jeans', again 107. I tried different approaches, this worked very well. I'm guessing these symbols are parsed on the server anyway, so that's why I decided to remove '+'

**Review by AleksandrsKondratjevs — APPROVED** [review]

File: packages/scandipwa/src/query/ProductList.query.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ export class ProductListQuery {
             },
             search: {
                 type: 'String!',
-                handler: (option) => option.replace(/\+/g, ' ')
+                handler: (option) => option.replace(/\+/g, '-')
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/query/ProductList.query.js:124]: No, stay with +. Is something wrong on our side.

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Comment by aleksandrakorolova** [comment]
1. Don't show 'true' or 'false' in the console 
Screenshot 2022-03-22 at 08 48 57
2. '-' is searched instead '+'
Screenshot 2022-03-22 at 09 05 08

**Comment by florianf22** [comment]
@AleksandrsKondratjevs Please check this. As I said, "+" will only be changed if search terms contains only and only "+" (it'll be changed with "-" and no results will be returned). I couldn't find anything related to why + might throw an error if any other symbol don't.

**Comment by aleksandrakorolova** [comment]
1. Fixed
2. Didn't fix

**Comment by leonazava** [comment]
This is a backend issue. This PR can be closed, I've linked a nw PR that tackles backend modules

**Pull request closed without merging**